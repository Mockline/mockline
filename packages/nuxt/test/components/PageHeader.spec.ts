import { describe, it, expect } from 'vitest'
import type {TypeOf} from "zod";
// @ts-ignore
import { MPageHeader } from '#components'
import ComponentRender from "../component-render";

describe('PageHeader', () => {
  it.each([
    ['base case', {}],
    ['all props', { props: { headline: 'healine', icon: 'heroicons-check', title: 'Title', links: [{ label: 'first', href: '/path/to/first' }], description: 'descr'}}],
    ['<MPageHeader title="titre"></MPageHeader>']
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof MPageHeader.props>) => {
    const html = await ComponentRender(nameOrHtml, options, MPageHeader)
    expect(html).toMatchSnapshot()
  })
})
