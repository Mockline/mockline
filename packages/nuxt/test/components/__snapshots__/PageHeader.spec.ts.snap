// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageHeader > renders <MPageHeader title="titre"></MPageHeader> correctly 1`] = `
"<div class="relative border-b border-(--color-border-primary) py-8">
  <!--v-if-->
  <div class="flex flex-col items-start gap-6 lg:flex-row">
    <!--v-if-->
    <div class="flex-1">
      <div>
        <h1 class="text-4xl font-bold mb-8 scroll-mt-[var(--ui-header-height)]">titre</h1>
      </div>
      <!--v-if-->
    </div>
  </div>
</div>"
`;

exports[`PageHeader > renders all props correctly 1`] = `
"<div class="relative border-b border-(--color-border-primary) py-8">
  <div>healine</div>
  <div class="flex flex-col items-start gap-6 lg:flex-row">
    <div><span class="text-primary size-10 flex-shrink-0">heroicons-check</span></div>
    <div class="flex-1">
      <div>
        <h1 class="text-4xl font-bold mb-8 scroll-mt-[var(--ui-header-height)]">Title</h1>
      </div>
      <p>descr</p>
    </div>
  </div>
</div>"
`;

exports[`PageHeader > renders base case correctly 1`] = `
"<div class="relative border-b border-(--color-border-primary) py-8">
  <!--v-if-->
  <div class="flex flex-col items-start gap-6 lg:flex-row">
    <!--v-if-->
    <div class="flex-1">
      <div>
        <h1 class="text-4xl font-bold mb-8 scroll-mt-[var(--ui-header-height)]"></h1>
      </div>
      <!--v-if-->
    </div>
  </div>
</div>"
`;
